manifests/components/oauth2-proxy.jsonnet

```diff
@@ -44,13 +44,36 @@ local OAUTH2_PROXY_IMAGE = (import "images.json")["oauth2_proxy"];
 
   hpa: kube.HorizontalPodAutoscaler($.p + "oauth2-proxy") + $.metadata {
     target: $.deploy,
-    spec+: {maxReplicas: 10},
+    spec+: {
+      // Put a cap on growth due to (eg) DoS attacks.
+      // Large sites will want to increase this to cover legitimate demand.
+      maxReplicas: 10,
+    },
+  },
+
+  pdb: kube.PodDisruptionBudget($.p + "oauth2-proxy") + $.metadata {
+    target_pod: $.deploy.spec.template,
+    spec+: {minAvailable: $.deploy.spec.replicas - 1},
   },
 
   deploy: kube.Deployment($.p + "oauth2-proxy") + $.metadata {
+    local this = self,
     spec+: {
+      // Ensure at least n+1.  NB: HPA will increase replicas dynamically.
+      replicas: 2,
       template+: {
         spec+: {
+          affinity+: {
```

Comment:
A comment that describes what this does would be very helpful.

Reply:
ditto.